src/viking.js

```diff
@@ -1,16 +1,108 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return 'Odin Owns You All!';
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      return 'A Saxon has died in combat';
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(viking) {
+    this.vikingArmy.push(viking);
+  }
+
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+
+  vikingAttack() {
+    let viking =
+      this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+
+    let saxon =
+      this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+
+    let saxonAttacked = saxon.receiveDamage(viking.strength);
+
+    if (saxon.health < 1) {
+      this.saxonArmy.splice(this.saxonArmy.indexOf(saxon), 1);
+    }
 
+    return saxonAttacked;
+  }
+
+  saxonAttack() {
+    let viking =
+      this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+
+    let saxon =
+      this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+
+    let vikingAttacked = viking.receiveDamage(saxon.strength);
+
+    if (viking.health < 1) {
+      this.vikingArmy.splice(this.vikingArmy.indexOf(viking), 1);
+    }
+
+    return vikingAttacked;
+  }
+}
+
+// Show Status
+function showStatus() {
+  if (this.saxonArmy.length < 1 && this.vikingArmy.length > 0) {
+    return 'Vikings have won the war of the century!';
+  } else if (this.vikingArmy.length < 1 && this.saxonArmy.length > 0) {
+    return 'Saxons have fought for their lives and survived another day...';
+  } else if (this.saxonArmy.length === 1 && this.vikingArmy.length === 1) {
+    return 'Vikings and Saxons are still in the thick of battle.';
+  }
+}
```

Comment:
Your `showStatus `method should be inside War class ;)

Reply:
Ahhh !!!  that makes sense now :D

Reply:
As always, thanks a lot for reviewing !! :)